File: ch01/examples/example-1.1.2.1/Main.hs
module Example where

main :: IO ()
main = do
  putStrLn $ "12 is prime: " <> show (isPrime 12)
  putStrLn $ "13 is prime: " <> show (isPrime 13)

isPrime :: Integer -> Bool
isPrime n = all (\k -> mod n k /= 0) [2 .. n - 1]

File: ch01/examples/example-1.1.2.1/main.ml
open Printf

let is_prime n =
  let numbers = List.init (n - 2) (fun i -> i + 2) in
  List.for_all (fun k -> n mod k <> 0) numbers

let () =
  printf "12 is prime: %b\n" (is_prime 12);
  printf "13 is prime: %b\n" (is_prime 13)
